### ISSUE TITLE
CPU jit tracing does not work with syncbatchnorm

### ISSUE THREAD

**Issue by abaqusstudent** [pytorch/pytorch#82464]
Using Pytorch 1.12 on Debian Linux (conda env).
SyncBatchNorm does not seem to work with jit.trace. On a mobilenet V3 model, I first do:

```
model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
```

After training, normally, with my models I do this:

```
 traced_script_module = torch.jit.trace(swa_model, (data))
 traced_script_module.save("./swa_model.tar")
```

both model and data are on cpu - the plan is to use the model for inference on cpu only.

However, it throws me this:

```
    raise ValueError("SyncBatchNorm expected input tensor to be on GPU")
ValueError: SyncBatchNorm expected input tensor to be on GPU
```

Is there a workaround for this? or is this some weird edge case which is not supported? 

Ofcourse, without SyncBatchNorm, everything is fine - ie the trace is produced and gives me the same result as the original model.

Thank you.

**Comment by davidberard98** [comment]
@abaqusstudent Does the error occur both with and without `torch.jit.trace`, or does it only occur when you `jit.trace` the model?

**Comment by abaqusstudent** [comment]
@davidberard98 yes, the problem is only when you jit trace the module. Else, everything is fine.
I think the problem is that syncbatchnorm only supports gpu (which I find a bit weird).
is there any known work around for this? I get really good results with syncBN, but am not able o trace it :sob:

**Comment by davidberard98** [comment]
Do you know how this works when it's unscripted? As you mention, SyncBatchNorm looks like it only supports GPU: won't this be an issue whether or not JIT tracing occurs?

**Comment by abaqusstudent** [comment]
@davidberard98 yes you are totally correct in that this is a problem for jit trace or even deployment. Thus, I am looking for a workaround.

in mmcv, there is this:
https://github.com/open-mmlab/mmcv/pull/1253

which is from here:
https://github.com/pytorch/pytorch/issues/41081#issuecomment-783961547

I dont know if this is the correct way to go about this? also, if this is correct why has this not been merged in pytorch? 
is this not the correct thing to do?

**Comment by evrimozmermer** [comment]
I found a workaround for this. Define the model, load the saved model and extract state dictionary, load state dictionary of saved model to defined model.
Example:
from yolo import Model
model = Model(...)
model_from_checkpoint = torch.load(..., map_location = "cpu")["model"]
model.load_state_dict(model_from_checkpoint.state_dict())
model.training=False
model.eval()
